python-frontend/src/main/java/org/sonar/python/semantic/v2/SymbolsModuleTypeProvider.java

```diff
@@ -137,15 +136,21 @@ private PythonType convertToFunctionType(FunctionSymbol symbol, Map<Symbol, Pyth
     return functionType;
   }
 
-  private static ParameterV2 convertParameter(FunctionSymbol.Parameter parameter) {
-    return new ParameterV2(parameter.name(),
-      PythonType.UNKNOWN,
-      parameter.hasDefaultValue(),
-      parameter.isKeywordOnly(),
-      parameter.isPositionalOnly(),
-      parameter.isKeywordVariadic(),
-      parameter.isPositionalVariadic(),
-      null);
+  private PythonType resolveReturnType(Map<Symbol, PythonType> createdTypesBySymbol, @Nullable ClassSymbol classSymbol) {
+    var returnType = PythonType.UNKNOWN;
+    if (classSymbol == null) {
+      return returnType;
+    }
+    if (rootModule == null) {
+      return convertToType(classSymbol, createdTypesBySymbol);
+    }
+    // If the symbol is a built-in symbol and the root module already exists, we search for it instead of recreating the symbol
+    // This is necessary to avoid duplicated symbols in the original ProjectSymbolTable to translate into duplicated PythonType
+    returnType = Optional.ofNullable(classSymbol.fullyQualifiedName())
```

Comment:
Note: I think this approach is a bit hacky, and very specific to the way we currently handle builtins (which are at the root of the project, whereas I think we should move them to a `builtins` package).

Somehow I feel like conversion from all symbols to types should first look into the `ProjectLevelTypeTable` to see if that symbol has already been created  (not only builtins), especially in the use case of converting a return type, which is not a place where a type is actually defined (i.e. it probably doesn't make sense to ever create new `PythonType` from here). Maybe implementing the `LazyType` resolution would make sense.